zio-schema/shared/src/main/scala/zio/schema/Diff.scala

```diff
@@ -254,6 +258,195 @@ object Differ {
+      case (Schema.Primitive(StandardType.Month), Diff.Temporal(distance: Long, ChronoUnit.MONTHS)) =>
+        Right((a: A) => Right(a.plus(distance)))
+      case (Schema.Primitive(StandardType.Duration(_)), Diff.Temporal(distance: Long, ChronoUnit.MILLIS)) =>
+        Right((a: A) => Right(JDuration.of(a.get(ChronoUnit.MILLIS) + distance, ChronoUnit.MILLIS)))
+
+      // TODO need to figure out how to deal with this -- which one to add regDiff or leapDiff?
+      case (Schema.Primitive(StandardType.MonthDay), Diff.MonthDays(regDiff, leapDiff @ _)) =>
+        Right((a: A) => Right(MonthDay.from(ChronoUnit.DAYS.addTo(a.atYear(2001), regDiff.toLong))))
+
+      case (s @ Schema.Lazy(_), diff) => patch(s.schema, diff)
+
+      case (Schema.Optional(_), Diff.Total(_, Diff.Tag.Left))      => Right((_: A) => Right(None))
+      case (Schema.Optional(_), Diff.Total(right, Diff.Tag.Right)) => Right((_: A) => Right(Some(right)))
+      case (Schema.Optional(schema), diff)                         => patch(schema, diff).map(f => (a: A) => f(a.get).map(Some(_)))
+
+      case (Schema.Tuple(leftSchema, rightSchema), Diff.Tuple(leftDiff, rightDiff)) => {
+        patch(leftSchema, leftDiff) -> patch(rightSchema, rightDiff) match {
+          case (Left(e1), Left(e2)) => Left(s"Errors: $e1 and $e2")
+          case (_, Left(e))         => Left(e)
+          case (Left(e), _)         => Left(e)
+          case (Right(f1), Right(f2)) =>
+            Right((a: A) => {
+              f1(a._1) -> f2(a._2) match {
+                case (Left(e1), Left(e2)) => Left(s"Errors: $e1 and $e2")
+                case (_, Left(e))         => Left(e)
+                case (Left(e), _)         => Left(e)
+                case (Right(l), Right(r)) => Right((l, r))
+              }
+            })
+        }
+      }
+
+      case (Schema.Sequence(schema, fromChunk, toChunk), Diff.Sequence(diffs)) =>
+        Right { (a: A) =>
+          diffs
+            .zipAll(toChunk(a))
+            .flatMap {
+              case (Some(Diff.Total(right, Diff.Tag.Right)), _) => Some(Right(right))
+              case (Some(Diff.Total(_, Diff.Tag.Left)), _)      => None
+              case (Some(diff), Some(value))                    => Some(patch(schema, diff).flatMap(f => f(value)))
+              case (None, Some(value))                          => Some(Left(s"Diff missing for value=$value."))
+              case (Some(diff), None)                           => Some(Left(s"Value missing for diff=$diff."))
+              case (None, None)                                 => Some(Left(s"Unknown error in sequence."))
+            }
+            .partitionMap(identity) match {
+            case (Chunk.empty, values) => Right(fromChunk(values))
+            case (errors, _)           => Left(s"Running patch produced the following error(s): ${errors.toList}.")
+          }
+        }
+
+      // TODO - how do you tell if diff is from left or right?
```

Comment:
I think we probably need to tag the original `Diff` with `Tag.Left` or `Tag.Right`.

Reply:
I created another Diff type to deal with eithers:
  `final case class Either(value: Diff, tag: Tag) extends Diff`

This is a little ugly, please let me know if you can think of a nicer way:
`      case (Schema.EitherSchema(leftSchema, _), Diff.Either(diff, Diff.Tag.Left)) =>
        patch(leftSchema, diff).map(
          f =>
            (a: A) =>
              Right(a.left.flatMap(f(_) match {
                case Right(value) => Left(value)
                case Left(value)  => Left(value)
              }))
        )
`
Thanks!

Reply:
Ugly and working is better than pretty and not working :)